state-reconstruct-fetcher/src/types/mod.rs

```diff
@@ -0,0 +1,106 @@
+use eyre::Result;
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+pub mod v1;
+pub mod v2;
+
+#[allow(clippy::enum_variant_names)]
+#[derive(Error, Debug)]
+pub enum ParseError {
+    #[error("invalid Calldata: {0}")]
+    InvalidCalldata(String),
+
+    #[error("invalid StoredBlockInfo: {0}")]
+    InvalidStoredBlockInfo(String),
+
+    #[error("invalid CommitBlockInfo: {0}")]
+    InvalidCommitBlockInfo(String),
+
+    #[allow(dead_code)]
+    #[error("invalid compressed bytecode: {0}")]
+    InvalidCompressedByteCode(String),
+
+    #[error("invalid compressed value: {0}")]
+    InvalidCompressedValue(String),
+}
+
+/// Block with all required fields extracted from a [`CommitBlockInfo`].
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CommitBlock {
+    /// L1 block number.
+    pub l1_block_number: Option<u64>,
+    /// L2 block number.
+    pub l2_block_number: u64,
+    /// The state root of the full state tree.
+    pub new_state_root: Vec<u8>,
+    /// Storage write access as a concatenation key-value.
+    pub initial_storage_changes: IndexMap<[u8; 32], [u8; 32]>,
+    /// Storage write access as a concatenation index-value.
+    pub repeated_storage_changes: IndexMap<u64, [u8; 32]>,
+    /// (contract bytecodes) array of L2 bytecodes that were deployed.
+    pub factory_deps: Vec<Vec<u8>>,
+}
+
+impl CommitBlock {
+    pub async fn from_commit_block(block_type: CommitBlockInfo) -> Self {
+        match block_type {
+            CommitBlockInfo::V1(block) => CommitBlock {
+                l1_block_number: None,
+                l2_block_number: block.block_number,
+                new_state_root: block.new_state_root,
+                initial_storage_changes: block.initial_storage_changes,
+                repeated_storage_changes: block.repeated_storage_changes,
+                factory_deps: block.factory_deps,
+            },
+            CommitBlockInfo::V2(_block) => todo!(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum CommitBlockInfo {
+    V1(v1::CommitBlockInfo),
+    V2(v2::CommitBlockInfo),
+}
+
+// TODO: Do we need this?
```

Comment:
Most probably not. If there is a need in the future, git would have the code so therefore I guess this can be zapped.